code/modules/antagonists/wizard/abilities/animal.dm

```diff
@@ -115,14 +118,18 @@ var/list/animal_spell_critter_paths = list(/mob/living/critter/small_animal/cat,
 		smoke.attach(target)
 		smoke.start()
 
-		if (target.mind && (target.mind.assigned_role != "Animal") || (!target.mind || !target.client))
+		if (!target.mind || !target.client)
 			boutput(target, SPAN_ALERT("<B>You feel your flesh painfully ripped apart and reformed into something else!</B>"))
-			if (target.mind)
-				target.mind.assigned_role = "Animal"
 			target.emote("scream", 0)
 
 			target.unequip_all()
-			var/mob/living/critter/C = target.make_critter(pick(animal_spell_critter_paths))
```

Comment:
Maybe better to add an argument to `make_critter` that causes it not to delete the original, rather than recreate it here? You're missing things like gender and bioeffect copying.